### ISSUE TITLE
[WARNING] Ignore page due to error: maximum recursion depth exceeded while calling a Python object

### ISSUE THREAD

**Issue by mdperezr13** [dothinking/pdf2docx#105]
Hi!!!

When generating a pdf of 15 pages if it generates all but when generating another one of 8 I get that warning and it eliminates page 3, with the old version of pdf2docx 

With the old version of the pdf2docx if it generates the 8 but in the case of the 15 page pdf it removes the 3

any suggestion
![1](https://user-images.githubusercontent.com/56566355/126402358-29adffd2-2fca-469a-8525-4cd0153d623a.JPG)
![2](https://user-images.githubusercontent.com/56566355/126402364-ffe72763-dfbf-435a-9afe-31abc280a560.JPG)

**Comment by dothinking** [comment]
Per the warning message, it seems that page 3 (of the 8 pages pdf) has a layout (vector graphics, guess) that the latest `pdf2docx` couldn't parse. 


> With the old version of the pdf2docx if it generates the 8 but in the case of the 15 page pdf it removes the 3

The vector graphic detecting method is changed in the latest version, so different behaviors as you faced. Anyway, it does have issue. It would be great if you can provide the sample pdf for further investigation.

**Comment by mariojaros** [comment]
Hi, I am dealing with a similar exception when I generating Docx file from pdf. I am using it as Python library in azure python function. I think that in pdf is some vector graphics
[DE000SF71D61_KID.pdf](https://github.com/dothinking/pdf2docx/files/7292402/DE000SF71D61_KID.pdf)
c 

screenshots from console:

<img width="909" alt="Screenshot 2021-10-06 at 11 21 53" src="https://user-images.githubusercontent.com/11440435/136175984-0f7faf93-5be2-44fe-9ad2-b27d2372166c.png">

<img width="1127" alt="Screenshot 2021-10-06 at 11 20 56" src="https://user-images.githubusercontent.com/11440435/136175819-0979a7d0-4197-41e8-a708-1fa5bfad6cf0.png">

But my problem is also that during the execution process is stopped and I don't get any pages from pdf.

**Comment by dothinking** [comment]
@mariojaros, thanks for providing the test file.

The root cause is discrete shapes (screenshot below) that makes the table layout parsing into endless loop until maximum recursion depth exceeded, as the error info said.


![image](https://user-images.githubusercontent.com/13885442/153718897-3413ecef-87ab-4878-bb0e-b756b49a4c87.png)


It was fixed by merging them together as a complete shape, then it could be parsed as table background successfully. This fix will be released per coming `v0.5.3`, or have a test per `dev` branch in advance.

```
$ pip install git+git://github.com/dothinking/pdf2docx.git@dev --upgrade
```

**Comment by yang-521** [comment]
I don't know why there was no problem when I used it in the development environment, but the error "maximum recursion depth exceeded" occurred after packaging. I used:  python3.13.2   win10  pdf2docx 0.5.8
